internal/core/config/monitor.go

```diff
@@ -129,8 +133,14 @@ func (mc *MonitorConfig) Hash() uint64 {
 	return hash
 }
 
+// GetExtraMetrics returns additional metrics that should be allowed through.
+func (mc *MonitorConfig) GetExtraMetrics() []string {
+	return nil
+}
+
 // MonitorCustomConfig represents monitor-specific configuration that doesn't
 // appear in the MonitorConfig struct.
 type MonitorCustomConfig interface {
 	MonitorConfigCore() *MonitorConfig
+	GetExtraMetrics() []string
```

Comment:
I think this would be a bit cleaner as a separate interface instead of reusing this one.  Then, the monitor's config types can do an interface assertion (the `var _ ... = ` thing) and get some compiler check that they are actually implementing the interface and make it clear to those reading the monitor the intended interface.

Also Go convention for getters is to not have a `Get` prefix.